clients/go/zbc/credentialsProvider_test.go

```diff
@@ -59,25 +84,154 @@ func TestZeebeClientCredentialsProvider(t *testing.T) {
 	if errorStatus, ok := status.FromError(err); ok {
 		require.Equal(t, codes.Unimplemented, errorStatus.Code())
 	}
-	require.Equal(t,
-		"Bearer expectedToken", interceptor.capturedToken)
+	require.Equal(t, tokenType+" "+accessToken, interceptor.capturedToken)
+}
+
+func TestOAuthProviderWithoutAuthzServerUrl(t *testing.T) {
+	// when
+	_, err := NewOAuthCredentialsProvider(&OAuthProviderConfig{
+		ClientId:     clientId,
+		ClientSecret: clientSecret,
+		Audience:     audience,
+	})
+
+	// then
+	require.EqualValues(t, InvalidArgumentError, errors.Cause(err))
+	require.EqualValues(t, invalidArgumentError("authorization server URL", "cannot be blank").Error(), err.Error())
+}
+
+func TestOAuthProviderWithMalformedAuthzServerUrl(t *testing.T) {
+	// when
+	_, err := NewOAuthCredentialsProvider(&OAuthProviderConfig{
+		ClientId:               clientId,
+		ClientSecret:           clientSecret,
+		Audience:               audience,
+		AuthorizationServerUrl: "foo",
+	})
+
+	// then
+	require.EqualValues(t, InvalidArgumentError, errors.Cause(err))
+	require.EqualValues(t, invalidArgumentError("authorization server URL", "must be a valid URL").Error(), err.Error())
+}
+
+func TestOAuthProviderWithoutClientId(t *testing.T) {
+	// when
+	_, err := NewOAuthCredentialsProvider(&OAuthProviderConfig{
+
+		ClientSecret:           clientSecret,
+		Audience:               audience,
+		AuthorizationServerUrl: "http://foo",
+	})
+
+	// then
+	require.EqualValues(t, InvalidArgumentError, errors.Cause(err))
+	require.EqualValues(t, invalidArgumentError("client ID", "cannot be blank").Error(), err.Error())
 }
 
-func TestZeebeClientCredentialsProviderWithEmptyPath(t *testing.T) {
+func TestOAuthProviderWithoutClientSecret(t *testing.T) {
 	// when
-	_, err := NewZeebeClientCredentialsProvider("")
+	_, err := NewOAuthCredentialsProvider(&OAuthProviderConfig{
+		ClientId:               clientId,
+		Audience:               audience,
+		AuthorizationServerUrl: "http://foo",
+	})
 
 	// then
-	require.Error(t, errors.Cause(err), InvalidPathError)
+	require.EqualValues(t, InvalidArgumentError, errors.Cause(err))
+	require.EqualValues(t, invalidArgumentError("client secret", "cannot be blank").Error(), err.Error())
 }
 
-func TestZeebeClientCredentialsProviderWithWrongPath(t *testing.T) {
+func TestOAuthProviderWithoutAudience(t *testing.T) {
 	// when
-	wrongPath := "../resources/creds_wrong_path"
-	_, err := NewZeebeClientCredentialsProvider(wrongPath)
+	_, err := NewOAuthCredentialsProvider(&OAuthProviderConfig{
```

Comment:
I think idiomatic Go is to use table tests for these kind of tests, like the validation one. We don't do those so far so I wouldn't insist on it since none of us has experience with them, but just a suggestion, we could try them.